src/execute.rs

```diff
@@ -8,6 +8,8 @@ use cw721::{Cw721ExecuteMsg, OwnerOfResponse};
 use cw721_base::helpers::Cw721Contract;
 use sg_std::Response;
 
+// TODO: there should be a fee for creating an offer, or it should be a contract that consumes gas.
+
```

Comment:
adding more context here, storing offers and data related to it consume blockchain space and needs to be properly charged, launchpad contracts take creation fees  so its harder to spam, usually validators should run with non zero gas-prices so all this storage is properly charged.

Reply:
Our current solution to this problem is to limit the amount of offers a user can create.
https://github.com/pegasuszone/pegasus/blob/main/src/msg.rs#L66

Reply:
This can easily be gamed by just transferring to new accounts if someone just wants to be clever about it.

Reply:
True. I think @MbBrainz's original vision for this was for it to have zero fees, but if this does end up being gamed we should definitely incorporate a listing fee as a SudoParam.
Maybe no fee for now & then a governance proposal to introduce fees if it becomes a problem?

Reply:
This could work too, if you have it as optional (if it is 0 do nothing) and can be updated later I think it might work.

Reply:
Sounds like a plan!

Reply:
What about making the contract consume gas? The idea of the contract is that it is a free simple utility. If one would charge a fee for using it, it might not be worth it and tbh it doesn't feel justified, for such a basic utility. If the contract just consumes gas though, it's it's probably not enough to discourage people from using it, but still prevents spamming.

Reply:
@jhernandezb What do you think?

Reply:
Yeah gas will be super low for creating an offer,  I haven't tried the app yet but I guess if you have a tx example we can see how much it takes.

Reply:
Only takes a few seconds to create an offer on the app. Send an offer to our testnet account [here](https://test.pz-l.ink/LMv81wB).

Reply:
<img width="423" alt="Screen Shot 2022-09-17 at 10 15 23 PM" src="https://user-images.githubusercontent.com/3452489/190885470-d4daf7e2-4199-4cf7-a3a0-baad57ae7999.png">
Just tried it with a 3v3 offer and took 1M gas which is significantly